src/components/Navbar/Language.tsx

```diff
@@ -1,33 +1,86 @@
-import React from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 import { useRouter } from 'next/router'
+import Image from 'next/image'
+import LanguageItem from '../LanguageItem'
 
 const NavbarLanguage = () => {
+  const [isMenuOpen, setMenuOpen] = useState<boolean>(false)
+  const [selectedLocale, setSelectedLocale] = useState<string>('en')
+  const flagRef = useRef<HTMLDivElement>(null)
   const router = useRouter()
+  const { locales } = router
 
-  const handleLocaleClick = (locale: string) => {
+  useEffect(() => {
+    document.addEventListener('mousedown', handleMouseClick)
+
+    return () => {
+      document.removeEventListener('mousedown', handleMouseClick)
+    }
+  }, [])
+
+  const handleDropdownToggle = () => {
+    setMenuOpen((prevState) => !prevState)
+  }
+
+  const handleMouseClick = (event: MouseEvent) => {
+    const isClickedOutside: boolean = !flagRef.current?.contains(event.target as Node)
```

Comment:
How about naming the variable as an `event` name?
```suggestion
    const isFocused: boolean = !flagRef.current?.contains(event.target as Node)
```